Chi-squared_test by Catherine Ann Celeste Quinlan

Based on the wikipedia example of workers occupations and areas  A,B,C and D

Pearsons chi squared test is used here  for Independence for categorical variables .

In [1]:
import pandas as pd
from scipy.stats import chi2
from scipy.stats import chi2_contingency

table = {'Workers': ['White Collar','Blue Collar','No collar'],
         
        'A': [90,30,30],
         
        'B': [60,50,40],
        'C': [104,51,45],
        'D': [95,20,35]
        }


df = pd.DataFrame(table, columns = ['Workers', 'A','B','C','D'])

print (df)


        Workers   A   B    C   D
0  White Collar  90  60  104  95
1   Blue Collar  30  50   51  20
2     No collar  30  40   45  35


In [2]:
table2 =    [[90,30, 30],
			[60,50,40],
            [104,51,45],
             [95,20,35]]




In [3]:
chi2value, p, dof, ex = (chi2_contingency(table2))# It returns a chi2value,a p value,and an estimate of degrees of freedom  before printing expected frequency  table

In [5]:
print('chi2value=%.1f' %chi2value)
print ('p=%.2f' %p)
print('dof=%.2f' % dof)
print(ex)

chi2value=24.6
p=0.00
dof=6.00
[[ 80.53846154  34.84615385  34.61538462]
 [ 80.53846154  34.84615385  34.61538462]
 [107.38461538  46.46153846  46.15384615]
 [ 80.53846154  34.84615385  34.61538462]]


As required we see a degree of freedom of 6 while our chi squared value is  rounded to 24.6

We use the degrees of freedom and a probability of 95%

In [ ]:

Hypothesis testing :







In [ ]:
We use the degrees of freedom and a probability of 95%

[1] https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.chi2_contingency.html

[2]https://machinelearningmastery.com/chi-squared-test-for-machine-learning/

https://www.statisticshowto.com/probability-and-statistics/chi-square/